sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_polling.py

```diff
@@ -6,53 +6,52 @@
 import time
 
 from azure.core.polling import PollingMethod
-from azure.keyvault.certificates._shared import parse_vault_id
 
 logger = logging.getLogger(__name__)
 
 
 class CreateCertificatePoller(PollingMethod):
-    def __init__(self, interval=5, unknown_issuer=False):
+    def __init__(self, get_certificate_command, unknown_issuer=False, interval=5):
         self._command = None
-        self._status = None
-        self._certificate_id = None
-        self.polling_interval = interval
-        self.unknown_issuer = unknown_issuer
+        self._resource = None
+        self._pending_certificate_op = None
+        self._get_certificate_command = get_certificate_command
+        self._unknown_issuer = unknown_issuer
+        self._polling_interval = interval
 
     def _update_status(self):
         # type: () -> None
-        pending_certificate = self._command()
-        self._status = pending_certificate.status.lower()
-        if not self._certificate_id:
-            self._certificate_id = parse_vault_id(pending_certificate.id)
+        self._pending_certificate_op = self._command()
 
     def initialize(self, client, initial_response, _):
         # type: (Any, Any, Callable) -> None
         self._command = client
-        self._status = initial_response
+        self._pending_certificate_op = initial_response
 
     def run(self):
         # type: () -> None
         try:
             while not self.finished():
                 self._update_status()
-                time.sleep(self.polling_interval)
+                time.sleep(self._polling_interval)
+            if self._pending_certificate_op.status.lower() == 'completed':
+                self._resource = self._get_certificate_command()
+            else:
+                self._resource = self._pending_certificate_op
         except Exception as e:
             logger.warning(str(e))
             raise
 
     def finished(self):
         # type: () -> bool
-        if self.unknown_issuer:
+        if self._unknown_issuer:
```

Comment:
`CertificateOperation` has an `issuer_name` property. Can we use that instead of having to know upfront whether the issuer is unknown?

Reply:
switched, thanks!